src/main/java/com/cloudbees/jenkins/support/impl/SlaveCommandStatistics.java

```diff
@@ -174,7 +199,25 @@ private void print(PrintWriter out) {
         public void preOnline(Computer c, Channel channel, FilePath root, TaskListener listener) throws IOException, InterruptedException {
             channel.addListener(ExtensionList.lookupSingleton(SlaveCommandStatistics.class).statistics.computeIfAbsent(c.getName(), k -> new Statistics()));
         }
-
     }
 
+    // Rotate Statistics entries between #statistics and #statLog separating the live entries from historical ones so we
+    // can put a capacity constrain on the latter.
```

Comment:
IMO it would suffice just to have to simple cap on the number of statistics kept, with some sort of LRU behavior. There is no particular reason that we _must_ keep statistics for a node just because it happens to still be defined. Someone reviewing the bundle is most likely just interested in statistics for recently active nodes.

Reply:
Yeah well, this is safer and already implemented so ...